From 6ede8c9ce215de50b8031ef38c49b6b7fae8e7ef Mon Sep 17 00:00:00 2001
From: Serguei Katkov <serguei.i.katkov@intel.com>
Date: Fri, 20 Sep 2013 15:47:08 +0700
Subject: Dalvik: allocateAuxStructures incorrectly check no memory condition

BZ: 139884

allocateAuxStructures checks the status of dvmAllocRegion invocation
expecting that it returns MAP_ERROR in case of error while it
returns NULL.

Category: device-enablement
Domain: AOSP-Dalvik-Runtime
Origin: internal
Upstream-Candidate: yes

Change-Id: I22e3cc329a635acfd5a846fef196a65f2386f081
Orig-MCG-Change-Id: I4b17498f533545a4248aa92cac38e3701ea8523d
Signed-off-by: Serguei Katkov <serguei.i.katkov@intel.com>
Reviewed-on: http://android.intel.com:8080/133306
Reviewed-by: Beyler, Jean Christophe <jean.christophe.beyler@intel.com>
Reviewed-by: Lupusoru, Razvan A <razvan.a.lupusoru@intel.com>
Reviewed-by: Banerji, Udayan <udayan.banerji@intel.com>
Reviewed-by: lab_aqa <lab_aqa@intel.com>
Reviewed-by: Popov, Ivan G <ivan.g.popov@intel.com>
Reviewed-by: Chen, Dong-Yuan <dong-yuan.chen@intel.com>
Tested-by: Chen, Dong-Yuan <dong-yuan.chen@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 vm/DvmDex.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/vm/DvmDex.cpp b/vm/DvmDex.cpp
index 28cd64e..f8aac5c 100644
--- a/vm/DvmDex.cpp
+++ b/vm/DvmDex.cpp
@@ -56,7 +56,7 @@ static DvmDex* allocateAuxStructures(DexFile* pDexFile)
 
     u1 *blob = (u1 *)dvmAllocRegion(totalSize,
                               PROT_READ | PROT_WRITE, "dalvik-aux-structure");
-    if ((void *)blob == MAP_FAILED)
+    if (blob == NULL)
         return NULL;
 
     pDvmDex = (DvmDex*)blob;
-- 
1.7.4.1

